crates/ra_parser/src/grammar/items.rs

```diff
@@ -100,9 +100,20 @@ pub(super) fn maybe_item(p: &mut Parser, m: Marker, flavor: ItemFlavor) -> Resul
 
     // test_err unsafe_block_in_mod
     // fn foo(){} unsafe { } fn bar(){}
-    if p.at(T![unsafe]) && p.nth(1) != T!['{'] {
-        p.eat(T![unsafe]);
-        has_mods = true;
+
+    // test_err unsafe_default_impl
+    // unsafe default impl Foo {}
+    if p.at(T![unsafe]) {
+        match p.nth(1) {
+            T!['{'] => {}
+            T![default] => {
+                p.err_and_bump("expected default unsafe, found unsafe default");
```

Comment:
@matklad any idea why this error never manifests.

Reply:
default is a contextual keyword, so you get an `IDENT` from lexer, and not `KW_DEFAUL` at this point.